src/main/java/org/carlspring/cloud/storage/s3fs/util/S3Utils.java

```diff
@@ -38,6 +39,44 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(S3Utils.class);
 
+    /**
+     * Returns a list of {@link S3Object}s for the given path.
+     *
+     * @param s3Path {@link S3Path}
+     * @return {@link List} of {@link S3Object}
+     * @throws NoSuchBucketException
+     *     The specified bucket does not exist.
+     * @throws SdkException
+     *    Base class for all exceptions that can be thrown by the SDK (both service and client). Can be used for
+     *    catch all scenarios.
+     * @throws SdkClientException
+     *     If any client side error occurs such as an IO related failure, failure to get credentials, etc.
+     * @throws S3Exception
+     *     Base class for all service exceptions.
+     */
+    public List<S3Object> listS3Objects(S3Path s3Path)
+    {
+
+        final String key = s3Path.getKey();
+        final String bucketName = s3Path.getFileStore().name();
+        final S3Client client = s3Path.getFileSystem().getClient();
+
+        // is a virtual directory
+        String keyFolder = key;
+        if (!keyFolder.endsWith("/"))
+        {
+            keyFolder += "/";
+        }
+
+        final ListObjectsV2Request request = ListObjectsV2Request.builder()
+            .bucket(bucketName)
+            .prefix(keyFolder)
+            .maxKeys(1)
+            .build();
```

Comment:
Cosmetic: Just code styke changes.
```suggestion
        final ListObjectsV2Request request = ListObjectsV2Request.builder()
                                                                 .bucket(bucketName)
                                                                 .prefix(keyFolder)
                                                                 .maxKeys(1)
                                                                 .build();
```